app/Libraries/Elasticsearch/Search.php

```diff
@@ -310,8 +311,15 @@ private function runQuery(string $operation, callable $callable)
                 $this->getDatadogTags()
             );
         } catch (ElasticsearchException $e) {
-            $this->error = $e;
-            $this->handleError($e, $operation);
+            $err = json_decode($e->getMessage(), true);
+
+            // handle cursor parsing error differently
+            if (is_array($err) && str_starts_with($err['error']['caused_by']['reason'] ?? '', 'Failed to parse search_after value for field ')) {
+                $this->error = new InvalidCursorException();
```

Comment:
probably better to move to `handleError` and have it return error to use (and set appropriate tag for datadog).